content/blog/2019-08-alphas.md

```diff
@@ -0,0 +1,231 @@
++++
+date = "2019-08-08"
+title = "Tokio alpha release with async & await"
+description = "August 08, 2019"
+menu = "blog"
+weight = 988
++++
+
+We're pleased to announce [the release][crates] of the first Tokio alpha with async / await
+support. This includes updating all of the Tokio crates to use `std::future`
+instead of `futures` 0.1. It also includes adding `async fn` versions of the
+APIs.
+
+Get it by adding:
+
+```toml
+tokio = "=0.2.0-alpha.1"
+```
+
+to your `Cargo.toml` file. This is how an echo server is now written:
+
+```rust,ignore
+#![feature(async_await)]
+
+use tokio::net::TcpListener;
+use tokio::prelude::*;
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let addr = "127.0.0.1:8080".parse()?;
+    let mut listener = TcpListener::bind(&addr)?;
+
+    loop {
+        let (mut socket, _) = listener.accept().await?;
+
+        tokio::spawn(async move {
+            let mut buf = [0; 1024];
+
+            // In a loop, read data from the socket and write the data back.
+            loop {
+                let n = match socket.read(&mut buf).await {
+                    // socket closed
+                    Ok(n) if n == 0 => return,
+                    Ok(n) => n,
+                    Err(e) => {
+                        println!("failed to read from socket; err = {:?}", e);
+                        return;
+                    }
+                };
+
+                // Write the data back
+                if let Err(e) = socket.write_all(&buf[0..n]).await {
+                    println!("failed to write to socket; err = {:?}", e);
+                    return;
+                }
+            }
+        });
+    }
+}
+```
+
+More examples are in the [git repository][examples];
+
+# The road so far
+
+Tokio was first announced almost exactly [three years
+ago][tokio-ann]. When Tokio was first released, it contained a minimal, single
+threaded, scheduler and APIs for TCP. Since then, it has grown to include a
+multi-threaded work-stealing scheduler, APIs for timers, TCP, UDP, UDS, file
+system access, signal handling, managing child processes, high performance async
+aware channels, and more. All this thanks to a [dedicated team of
+maintainers][maint] and an [active community of contributors][contrib]. Tokio
+would not be where it is without you! Thank you so much.
+
+The goal was (and still is) to provide an ergonomic library for writing
+asynchronous applications for Rust without compromising on performance. The past
+three years has been spent building out the Rust I/O ecosystem and discovering
+idioms and patterns for writing idiomatic asynchronous Rust. Over the past three
```

Comment:
TIOLI: "idioms and patterns for writing idiomatic..." seems a little redundant to me.